ratis-netty/src/main/java/org/apache/ratis/netty/server/NettyServerStreamRpc.java

```diff
@@ -203,39 +206,43 @@ StreamInfo get(Key key) {
   private final EventLoopGroup workerGroup = new NioEventLoopGroup();
   private final ChannelFuture channelFuture;
 
-  private final StateMachine stateMachine;
   private final StreamMap streams = new StreamMap();
+  private final ConcurrentMap<RaftGroupId, StateMachine> stateMachines;
+  private final StateMachine.Registry stateMachineRegistry;
```

Comment:
It should get stateMachine from RaftServer.  I have filed https://issues.apache.org/jira/browse/RATIS-1124 since the change is quite involved.